packages/openapi3/src/types.ts

```diff
@@ -695,3 +699,391 @@ export interface Ref<T> {
 }
 
 export type Refable<T> = Ref<T> | T;
+
+export type JsonSchemaType =
+  | "array"
+  | "boolean"
+  | "integer"
+  | "null"
+  | "number"
+  | "object"
+  | "string";
+
+export type JsonSchema<AdditionalVocabularies extends {} = {}> = AdditionalVocabularies & {
```

Comment:
I'm hoping to eventually move `JsonSchema` to the `@typespec/json-schema` package, so for `JsonSchema` I wanted it to only match what's in the json schema 2020-12 spec. You can imagine then that the json-schema emitter wouldn't need to care about any additional fields.

Open API 3.1 does add additional vocabulary to the schema - fields like `discriminator` and `xml` for example. I couldn't do a simple `interface OpenAPISchema3_1 extends JsonSchema` because the extra fields need to be available in sub-schemas as well (like additionalProperties, allOf/oneOf/anyOf, etc)

Reply:
What I'm curious about is that `AdditionalVocabularies` is not imported from external package, or init in this file, how can we use it?

Reply:
`AditionalVocabularies` is a generic type variable, so it gets set to whatever is passed into `JsonSchema<>`. The `OpenAPISchema3_1` type passes in the Open API 3.1 specific fields as the generic type.

`AdditionalVocabularies` also has a default value of an empty object so it is an optional type variable.

Reply:
Oh, right. I used to naming convension for generic type `TSomeType` that starts with `T` TT